<a href="https://colab.research.google.com/github/jihannaurah/BOOTCAMP/blob/main/MINI_PROJECT_BOOTCAMP_DIBIMBING.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
import pandas as pd
df = pd.read_csv('data.csv', encoding='latin1')
df.head()

,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,12/1/2010 8:26,2.55,17850.0,United Kingdom
1,536365,71053,WHITE METAL LANTERN,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,12/1/2010 8:26,2.75,17850.0,United Kingdom
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,12/1/2010 8:26,3.39,17850.0,United Kingdom


DATA QUALITY CHECK

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    541909 non-null  object 
 1   StockCode    541909 non-null  object 
 2   Description  540455 non-null  object 
 3   Quantity     541909 non-null  int64  
 4   InvoiceDate  541909 non-null  object 
 5   UnitPrice    541909 non-null  float64
 6   CustomerID   406829 non-null  float64
 7   Country      541909 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 33.1+ MB


In [5]:
df.isnull().sum()

,0
InvoiceNo,0
StockCode,0
Description,1454
Quantity,0
InvoiceDate,0
UnitPrice,0
CustomerID,135080
Country,0


In [6]:
df.duplicated().sum()

np.int64(5268)

In [7]:
df[df.duplicated()]

,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country
517,536409,21866,UNION JACK FLAG LUGGAGE TAG,1,12/1/2010 11:45,1.25,17908.0,United Kingdom
527,536409,22866,HAND WARMER SCOTTY DOG DESIGN,1,12/1/2010 11:45,2.10,17908.0,United Kingdom
537,536409,22900,SET 2 TEA TOWELS I LOVE LONDON,1,12/1/2010 11:45,2.95,17908.0,United Kingdom
539,536409,22111,SCOTTIE DOG HOT WATER BOTTLE,1,12/1/2010 11:45,4.95,17908.0,United Kingdom
555,536412,22327,ROUND SNACK BOXES SET OF 4 SKULLS,1,12/1/2010 11:49,2.95,17920.0,United Kingdom
...,...,...,...,...,...,...,...,...
541675,581538,22068,BLACK PIRATE TREASURE CHEST,1,12/9/2011 11:34,0.39,14446.0,United Kingdom
541689,581538,23318,BOX OF 6 MINI VINTAGE CRACKERS,1,12/9/2011 11:34,2.49,14446.0,United Kingdom
541692,581538,22992,REVOLVER WOODEN RULER,1,12/9/2011 11:34,1.95,14446.0,United Kingdom
541699,581538,22694,WICKER STAR,1,12/9/2011 11:34,2.10,14446.0,United Kingdom


DATA CLEANING

In [8]:
df = df.drop_duplicates()

In [9]:
df = df.dropna(subset=['CustomerID'])

In [10]:
print(df.shape)

(401604, 8)


In [11]:
df.isnull().sum()

,0
InvoiceNo,0
StockCode,0
Description,0
Quantity,0
InvoiceDate,0
UnitPrice,0
CustomerID,0
Country,0


In [12]:
df = df.reset_index(drop=True)

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 401604 entries, 0 to 401603
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    401604 non-null  object 
 1   StockCode    401604 non-null  object 
 2   Description  401604 non-null  object 
 3   Quantity     401604 non-null  int64  
 4   InvoiceDate  401604 non-null  object 
 5   UnitPrice    401604 non-null  float64
 6   CustomerID   401604 non-null  float64
 7   Country      401604 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 24.5+ MB


FILTERING "QUANTITY, UNITPRICE"

In [14]:
df = df[df['Quantity'] > 0]

In [15]:
df = df[df['UnitPrice'] > 0]

In [16]:
df = df.reset_index(drop=True)

In [17]:
print("Hasil:")
print(df.shape)

Hasil:
(392692, 8)


In [18]:
print("\ninfo detail:")
df.info()


info detail:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 392692 entries, 0 to 392691
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    392692 non-null  object 
 1   StockCode    392692 non-null  object 
 2   Description  392692 non-null  object 
 3   Quantity     392692 non-null  int64  
 4   InvoiceDate  392692 non-null  object 
 5   UnitPrice    392692 non-null  float64
 6   CustomerID   392692 non-null  float64
 7   Country      392692 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 24.0+ MB


MASKING

In [19]:
df['CustomerID'] = df['CustomerID'].astype(int).astype(str)
df['CustomerID'] = df['CustomerID'].str[:2] + '***'

df['StockCode'] = df['StockCode'].astype(str)
df['StockCode'] = df['StockCode'].str[:2] + '***'

print("Data Masking")
df.sample(20)

Data Masking


,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country
348344,577121,22***,POPCORN HOLDER,12,11/17/2011 18:29,0.85,12***,France
384575,580727,22***,LIPSTICK PEN FUSCHIA,12,12/5/2011 17:17,0.83,14***,United Kingdom
191886,561195,22***,BOUDOIR SQUARE TISSUE BOX,4,7/25/2011 13:57,0.39,14***,United Kingdom
375358,579846,22***,PAPER CHAIN KIT 50'S CHRISTMAS,18,11/30/2011 16:22,2.95,12***,United Kingdom
235130,566285,84***,ENCHANTED BIRD COATHANGER 5 HOOK,2,9/11/2011 14:51,4.65,13***,United Kingdom
218155,564199,22***,METAL SIGN TAKE IT OR LEAVE IT,3,8/23/2011 17:31,2.95,13***,United Kingdom
254549,568193,22***,METAL SIGN NEIGHBOURHOOD WITCH,5,9/25/2011 14:37,2.10,17***,United Kingdom
268820,569546,23***,TRIPLE HOOK ANTIQUE IVORY ROSE,1,10/4/2011 16:39,3.29,15***,United Kingdom
346,536401,20***,STRAWBERRY BATH SPONGE,1,12/1/2010 11:21,1.25,15***,United Kingdom
99497,549313,22***,LUNCH BAG SUKI DESIGN,30,4/8/2011 9:45,1.65,12***,Australia


**DATA DICTIONARY**
| Nama Kolom | Tipe | Penjelasan Singkat | Contohnya |
| :--- | :--- | :--- | :--- |
| **InvoiceNo** | Teks | Kode transaksi. Kalau ada huruf 'C' di depan, berarti itu barang yang dicancel/retur. | 536365 |
| **StockCode** | Teks | Kode buat tiap barang. (masking). | 85*** |
| **Description** | Teks | Nama barang atau penjelasan produk yang dibeli pelanggan. | BOX OF 6 MINI VINTAGE CRACKERS... |
| **Quantity** | Angka | Jumlah barang yang dibeli. Udah di filter jadi cuma angka yang di atas 0. | 6 |
| **InvoiceDate** | Teks | Tanggal sama jam kapan transaksi itu terjadi. | 12/1/2010 8:26 |
| **UnitPrice** | Desimal | Harga satu barang dalam Poundsterling (£). | 2.55 |
| **CustomerID** | Teks | ID Pelanggan. Udah di sensor (masking) juga buat jaga data sensitif. | 17*** |
| **Country** | Teks | Negara tempat belanja atau asal pelanggannya. | United Kingdom |

In [25]:
df.to_csv('Online_Retail_E-Commerce.csv', index=False)
df.to_excel('Online_Retail_E-Commerce.xlsx', index=False)
print('Tuntas!')

Tuntas!
